[Joy] Add "Quick start" and "Tutorial" pages (#32383)
mui/material-ui: docs/data/joy/getting-started/quick-start/quick-start.md

> # Quick start

Reviewer: I'm wondering if we shouldn't call this the day-to-day "Get started" page? Then, we can naturally continue to tailor the message as in "Joy is easy to get started with".

Reviewer: Thoughts about that? @siriwatknp

<p class="description">Get started with Joy, a simple and beautiful starting point for your design system, in 3 quick steps.</p>

## 1. Installation

Install Joy via either npm or yarn using the commands below in your directory of choice.

### npm

```sh
npm init react-app spark-joy
cd spark-joy
npm install @mui/joy @emotion/react @emotion/styled
npm start
```

> ### Yarn

Reviewer: ```suggestion
### yarn
```

```sh
yarn create react-app spark-joy
cd spark-joy
yarn add @mui/joy @emotion/react @emotion/styled
yarn start
```

> ## 2. Set up the `CssVarsProvider`

Reviewer: Feel like we're missing a word explaining what `CssVarsProvider` is. 

```suggestion
## 2. Set up the `CssVarsProvider` utility
```

Would that be correct? Something like that.

Author: I would not add `utility` here because I feel that it refers to as optional but `CssVarsProvider` is required.

Or to make it more generic, we could use `2. Set up the Provider`.

Reviewer: Would that be correct then? 
```suggestion
## 2. Set up the `CssVarsProvider` component
```

Author: That works.

> Open `src/App.js` and replace it with the code snippet below. You should see the text `Hello from Joy` being rendered on your browser.

Reviewer: ```suggestion
Go to your `App.js` file and replace it with the code snippet below. You should see the text `Hello from Joy` being rendered on your browser.
```

```jsx
import { CssVarsProvider } from '@mui/joy/styles';

function App() {
  return <CssVarsProvider>Hello from Joy</CssVarsProvider>;
}

export default App;
```

## 3. Render your first component

> Every Joy component needs to render under the `CssVarProvider` element. Let's start off with a button component.

Reviewer: ```suggestion
To render any Joy component, make sure you place them inside the `CssVarProvider` element. Let's start off with a button.
```
